src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+            else
+            {
+                starting_vd_edge = edge;
+            }
+            first = false;
+            continue;
+        }
+        first = false;
+    }
+    
+    assert(starting_vd_edge && ending_vd_edge);
+    assert(starting_vd_edge != ending_vd_edge);
+    
+    start_source_point = source_segment.to();
+    end_source_point = source_segment.from();
+}
+
+SkeletalTrapezoidation::SkeletalTrapezoidation(const Polygons& polys, float transitioning_angle
+, coord_t discretization_step_size
+, coord_t transition_filter_dist
+, coord_t beading_propagation_transition_dist
+)
+: polys(polys)
+, transitioning_angle(transitioning_angle)
+, discretization_step_size(discretization_step_size)
+, transition_filter_dist(transition_filter_dist)
+, beading_propagation_transition_dist(beading_propagation_transition_dist)
+{
+    init();
+}
+
+void SkeletalTrapezoidation::init()
+{
+    std::vector<Point> points; // remains empty
+
+    std::vector<Segment> segments;
+    for (size_t poly_idx = 0; poly_idx < polys.size(); poly_idx++)
+    {
+        ConstPolygonRef poly = polys[poly_idx];
+        for (size_t point_idx = 0; point_idx < poly.size(); point_idx++)
+        {
+            segments.emplace_back(&polys, poly_idx, point_idx);
+        }
+    }
+
+    vd_t vd;
+    construct_voronoi(points.begin(), points.end(),
+                                        segments.begin(), segments.end(),
+                                        &vd);
+
+#ifdef DEBUG
+    VoronoiUtils::debugOutput("output/vd.svg", vd, points, segments);
+#endif
+
+    for (const vd_t::edge_type& edge : vd.edges())
+    {
+        assert(edge.vertex0() == edge.twin()->vertex1());
+        assert(edge.vertex1() == edge.twin()->vertex0());
+        assert(edge.vertex1() == edge.next()->vertex0());
+        assert(edge.vertex0() == edge.prev()->vertex1());
+    }
+
+    
+    for (vd_t::cell_type cell : vd.cells())
+    {
+        if (!cell.incident_edge())
+        { // there is no spoon
+            continue;
+        }
+        Point start_source_point;
+        Point end_source_point;
+        vd_t::edge_type* starting_vd_edge = nullptr;
+        vd_t::edge_type* ending_vd_edge = nullptr;
+        // compute and store result in above variables
+        
+        if (cell.contains_point())
+        {
+            bool keep_going = computePointCellRange(cell, start_source_point, end_source_point, starting_vd_edge, ending_vd_edge, points, segments);
+            if (!keep_going)
+            {
+                continue;
+            }
+        }
+        else
+        {
+            computeSegmentCellRange(cell, start_source_point, end_source_point, starting_vd_edge, ending_vd_edge, points, segments);
+        }
+        
+        if (!starting_vd_edge || !ending_vd_edge)
+        {
+            assert(false && "each cell should start / end in a polygon vertex");
+            continue;
+        }
+        
+        // copy start to end edge to graph
```

Comment:
? Sentence? Are we copying twice, or...?

Reply:
Test reply! If github works properly you should see his!